middleware/index.ts

```diff
@@ -1,69 +1,185 @@
+              const msg = `Failed to send response [${this.id}]: ${result.errors.join(',')}`;
+              this.logger.error(msg);
+              throw new Error(msg);
+          } else {
+              return result;
+          }
+      } else {
+          throw new Error('Authorization not provided');
       }
-    },
-    "payload": {
-      "jobId": "PDF-JOB-2c33a77f-36c3-4c8a-aa9e-9a2925487349",
-      "url": "http://c.dev.visual.localhost.soma.force.com:6109/apex/advancedVPage?__pdffx-renderas-override__=true",
-      "html": null,
-      "isLightning": false,
-      "sessionId": "00Dxx0000006KYV!AQEAQFs8bC0sxQ_B.Qpt1c_GPLLEsQi9WGbCEUJlSJ8.MGC_Bzr1YQLu6m8vpW0VfEixurFuG8AhdYWpqSwdHrnhwLgT0681",
-      "lightningSessionId": null,
-      "requestVerificationToken": "AAAAAW5rqoZdAAAAAAAAAAAAAAAAAAAAAAAA4HheNFAbDl8vpenXOXb8kopR4X3uTxj9yZrtpnrW6j0RQzBQh6lr42mq_0OOuGMxDJgZPfJNLARHSAfkNWsZaFs="
-    },
-    "sfContext": {
-      "accessToken": "00Dxx0000006KYV!AQEAQHgHXd4DkZjmed1wSZMh85jXq8nAzAIt1zeK.NrvrHaSoemaPjbIDwfTWxdDrlu3Mpuft5JchelVkJ3mkwW_GKz2eOK.",
-      "functionInvocationId": "9mdxx00000002Hx",
-      "functionName": "salesforce/lightning-bridge/pdf-creator",
-      "requestId": "4SRXEGsFmeIAEV2VI5V21V",
-      "resource": null
+  }
+}
+
+/**
+ * Construct Event from invocation request.
+ *
+ * @param data    -- function payload
+ * @param headers -- request headers
+ * @param payload -- request payload
+ * @return event
+ */
+function createEvent(data: any, headers: any, payload: any): InvocationEvent {
+    return new InvocationEvent(
+        data,
+        payload.contentType,
+        payload.schemaURL,
+        payload.id,
+        payload.source,
+        payload.time,
+        payload.type,
+        headersToMap(headers)
+    );
+}
+
+function headersToMap(headers: any = {}): ReadonlyMap<string, ReadonlyArray<string>> {
+    const headersMap: Map<string, ReadonlyArray<string>> = new Map(Object.entries(headers));
+    return headersMap;
+}
+
+/**
+ * Construct User object from the request context.
+ *
+ * @param userContext -- userContext object representing invoking org and user
+ * @return user
+ */
+function createUser(userContext: any): User {
+    return new User(
+        userContext.userId,
+        userContext.username,
+        userContext.onBehalfOfUserId
+    );
+}
+
+/**
+ * Construct Org object from the request context.
+ *
+ * @param reqContext
+ * @return org
+ */
+function createOrg(logger: Logger, reqContext: any, accessToken?: string): Org {
+    const userContext = reqContext.userContext;
+    if (!userContext) {
+        const message = `UserContext not provided: ${JSON.stringify(reqContext)}`;
+        throw new Error(message);
+    }
+
+    const apiVersion = reqContext.apiVersion || process.env.FX_API_VERSION || Constants.CURRENT_API_VERSION;
+    const user = createUser(userContext);
+
+    // If accessToken was provided, setup APIs.
+    let dataApi: DataApi;
```

Comment:
I think these need to be of type `DataApi | undefined` and `UnitOfWork | undefined` to be able to be compiled strict.